staging/src/k8s.io/component-base/metrics/prometheus/version/metrics.go

```diff
@@ -29,7 +29,7 @@ var (
 			Help:           "A metric with a constant '1' value labeled by major, minor, git version, git commit, git tree state, build date, Go version, and compiler from which Kubernetes was built, and platform on which it is running.",
 			StabilityLevel: metrics.ALPHA,
 		},
-		[]string{"major", "minor", "gitVersion", "gitCommit", "gitTreeState", "buildDate", "goVersion", "compiler", "platform"},
```

Comment:
If we are going to make backwards compatible changes on this metric, we should try to parse out and add patch version too.

Reply:
Yeah, I understand now. It's a little bit more complicated. Can we done this before move it to stable? 

Speaking of the patch version you mentioned, there is a comment thinking of deprecating `gitMajor` and `gitMinor`:
https://github.com/kubernetes/kubernetes/blob/d62762f09090765cc3ff65fa113ec735f0c95327/staging/src/k8s.io/component-base/version/base.go#L37-L43

I'm not sure if the deprecating things still the right approach or we should add a new `gitPatch` fields.

Reply:
actually i think we need to revert this change. otherwise, we are going to end up emitting an inconsistent version metric from one in client-go (which we cannot easily change).

Reply:
nvm i was misremembering how this was wired up